### ISSUE TITLE
Eliminate use of `SubfieldBase` from StreamField

### ISSUE THREAD

**Issue by ironhouzi** [torchbox/wagtail#2070]
Some 3rd party django apps are using `SubfieldBase`, which are being depricated after Django 1.9 I believe.

Please try to get these fixed upstream.

**Comment by gasman** [comment]
Hi - I assume you're asking us to remove the occurrence of `SubfieldBase` from Wagtail, not to track down other third-party apps that use it? (Have updated issue title accordingly...)

**Comment by ironhouzi** [comment]
I assumed that your latest release with support for Django 1.9 already had ensured that these things had been weeded out from the Wagtail code base, but since wagtail relies so heavily on a handful of 3rd party Django apps, I assumed that some of these weren't updated with 1.9 support..

I also have not yet made use of any StreamField's, so I should perhaps provide some more information.

`news/models.py`:

``` python
from django.db import models
from django.core.paginator import Paginator, PageNotAnInteger, EmptyPage

from wagtail.wagtailcore.models import Page
from wagtail.wagtailcore.fields import RichTextField

from wagtail.wagtailimages.edit_handlers import ImageChooserPanel
from wagtail.wagtailadmin.edit_handlers import FieldPanel


class NewsEntryIndex(Page):
    @property
    def news(self):
        return NewsEntry.objects.live().descandant_of(self).order_by('-date')

    def get_context(self, request):
        # pagination
        page = request.GET.get('page')
        paginator = Paginator(self.news, 10)

        try:
            news = paginator.page(page)
        except PageNotAnInteger:
            news = paginator.page(1)
        except EmptyPage:
            news = paginator.page(paginator.num_pages)

        context = super().get_context(request)
        context['news'] = news

        return context

    class Meta:
        verbose_name = 'Nyhetsoversikt'
        verbose_name_plural = 'Nyhetsoversikter'

# NewsEntryIndex.content_panels = []
NewsEntryIndex.promote_panels = Page.promote_panels


class NewsEntry(Page):
    description = models.CharField('Beskrivelse', max_length=140)
    details = RichTextField('Detaljer', null=True, blank=True)
    date = models.DateField('Dato', auto_now=True, blank=False)
    feed_image = models.ForeignKey(
        'wagtailimages.Image',
        verbose_name='Bilde til nyhetsstrømmen',
        null=True,
        blank=True,
        on_delete=models.SET_NULL,
        related_name='+'
    )
    # event = models.ForeignKey(...)

    @property
    def news_index(self):
        return self.get_ancestors().type(NewsEntryIndex).last()

    content_panels = Page.content_panels + [
        FieldPanel('description'),
        FieldPanel('details'),
    ]

    promote_panels = Page.promote_panels + [
        ImageChooserPanel('feed_image'),
    ]

    def __str__(self):
        return self.description

    class Meta:
        verbose_name = 'Nyhet'
        verbose_name_plural = 'Nyheter'
```
`requirements.txt`:

```
beautifulsoup4==4.4.1
dj-database-url==0.3.0
Django==1.9
django-appconf==1.0.1
django-compressor==1.6
django-modelcluster==1.1
django-taggit==0.17.6
django-treebeard==3.0
djangorestframework==3.3.2
html5lib==0.9999999
Pillow==3.0.0
psycopg2==2.6.1
pytz==2015.7
six==1.10.0
Unidecode==0.4.18
wagtail==1.3
Willow==0.2.2
```

Either way, I would like to see the warnings go away when using Wagtail.

**Comment by gasman** [comment]
OK, thanks for the clarification! We'll deal with the dependencies when we come to implement Django 1.10 support, or hopefully sooner. It's quite possible that the relevant projects will implement the necessary fixes themselves in the meantime - but if not, we'll submit pull requests as required (as we've done for previous Django releases).

**Comment by FlipperPA** [comment]
I did some legwork on this, after seeing the deprecation warning. I'm running Django 1.9 with Wagtail 1.3. Grepping (-ri --include=*.py) my entire virtualenv only showed one occurrence in fields.py in class StreamField, so it would seem Wagtail's dependencies are good to go. Here's the line:

https://github.com/torchbox/wagtail/blob/master/wagtail/wagtailcore/fields.py#L44

I'd issue a P.R. but not sure on the best way to fix it, but I figured the sleuthing might help.
